### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by Lithnotep**
#Add users controller
#add users model
#add profile view
#add register view
#add user#new route
#add user#create route
#add all related specs

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  def create
+    user = User.new(user_params)
+    empty_params = user_params.select{|param, value| value.empty?}
+    if user.save
+      flash[:message] = "Hi #{user.name} you have registered and logged in!"
+      session[:user_id] = user.id
+      redirect_to "/profile"
+    elsif !empty_params.empty?
```
Comment (jpc20) [inline app/controllers/users_controller.rb:18]: Lets refactor this to the activerecord version when we get the chance

File: Gemfile [hunk 2]
```diff
@@ -13,6 +13,7 @@ gem 'sass-rails', '~> 5.0'
 gem 'uglifier', '>= 1.3.0'
 gem 'coffee-rails', '~> 4.2'
 gem 'jbuilder', '~> 2.5'
+gem 'bcrypt'
```
Comment (janegreene) [inline Gemfile:16]: used for password digests in future stories

File: spec/factories.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
```
Comment (Ashkanthegreat) [inline spec/factories.rb:1]: Let's figure out how to make this work

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  <%= text_field_tag :zip, @user[:zip] %><br/>
+
+  <%= label_tag "Email" %>
+  <%= text_field_tag :email %><br/>
+
+  <%= label_tag "Password" %><br/>
+  <%= password_field_tag :password =%><br/>
+
+  <%= label_tag "Confirm Password" %><br/>
+  <%= password_field_tag :confirm_password =%><br/>
```
Comment (jpc20) [inline app/views/users/new.html.erb:24]: We still need to find out how to check that passwords are the same

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<%= form_tag("/users", method: :post) do %>
+  <%= label_tag "Name" %>
+  <%= text_field_tag :name, @user[:name] %><br/>
```
Comment (janegreene) [inline app/views/users/new.html.erb:3]: this allows us to pre populate the inputs from previous attemps

**Pull request merged**